detector/zoo/mocked_detect.go

```diff
@@ -0,0 +1,82 @@
+package zoo
+
+import (
+	"errors"
+	"github.com/gogo/protobuf/proto"
+	log "github.com/golang/glog"
+	_ "github.com/mesos/mesos-go/mesosproto"
+	util "github.com/mesos/mesos-go/mesosutil"
+	"github.com/samuel/go-zookeeper/zk"
+)
+
+type MockMasterDetector struct {
+	*MasterDetector
+	conCh chan zk.Event
+	sesCh chan zk.Event
+}
+
+func NewMockMasterDetector(zkurls string) (*MockMasterDetector, error) {
+	log.V(4).Infoln("Creating mock zk master detector")
+	md, err := NewMasterDetector(zkurls)
+	if err != nil {
+		return nil, err
+	}
+	m := &MockMasterDetector{
+		MasterDetector: md,
+		conCh:          make(chan zk.Event, 5),
+		sesCh:          make(chan zk.Event, 5),
+	}
+
+	path := m.zkPath
+	connector := NewMockConnector()
+	connector.On("Children", path).Return([]string{"info_0", "info_5", "info_10"}, &zk.Stat{}, nil)
+	connector.On("Get", path).Return(m.makeMasterInfo(), &zk.Stat{}, nil)
+	connector.On("Close").Return(nil)
+	connector.On("ChildrenW", md.zkPath).Return([]string{m.zkPath}, &zk.Stat{}, (<-chan zk.Event)(m.sesCh), nil)
+
+	first := true
+	m.client.setFactory(asFactory(func() (Connector, <-chan zk.Event, error) {
+		if !first {
+			return nil, nil, errors.New("only 1 connector allowed")
+		} else {
+			first = false
+		}
+		return connector, m.conCh, nil
+	}))
+
+	return m, nil
+}
+
+func (m *MockMasterDetector) Start() {
```

Comment:
since this is a mock class, i'm not sure why we actually need a Start() func. I kept it because tests seemed to use it .. but not sure that it's really needed